File: 2023/rust/day-01/src/bin/sol.rs
use day_01::day1_sol;


fn main() {
    let input = "./input.txt";
    // let input = "./dummy_input.txt";
    let (sum_part1, sum_part2) = day1_sol(input);
    println!("Sum Part 1: {}", sum_part1);
    println!("Sum Part 2: {}", sum_part2);
}

File: 2024/python/day-01/sol.py
from collections import Counter

with open("input.txt") as f:
    location_ids = [tuple(map(int, line.split())) for line in f]

location_ids_l, location_ids_r = zip(*location_ids)

# Part 1: Calculate the sum of absolute differences
ans_1 = sum(
    abs(right - left)
    for left, right in zip(sorted(location_ids_l), sorted(location_ids_r))
)
print("Part 1:", ans_1)

# Part 2: Calculate the similarity score
rhs_frequency = Counter(location_ids_r)
similarity_score = sum(rhs_frequency[left] * left for left in location_ids_l)
print("Part 2:", similarity_score)
